### ISSUE TITLE
Model never learned to output sentence ending (eos) tokens due to tokenizer issue.

### ISSUE THREAD

**Issue by hungphongtrn** [janhq/research#13]
Due to `tokenizer.pad_token = tokenizer.eos_token` in the training code, the eos_token is recognized as the pad_token, which will then be converted to `-100` and be ignored during loss calculation. 
Sample input:
```dict
{
     'input_ids': tensor([[128000, 128000, 128006,  ...,     24,     13, 128009]], device='cuda:0'), 
     'attention_mask': tensor([[1, 1, 1,  ..., 1, 1, 1]], device='cuda:0'), 
     'labels': tensor([[128000, 128000, 128006,  ...,     24,     13,   -100],device='cuda:0')
}
```
The last token of input_ids 128009 is equivalent to <eos_token> and is converted to -100 as in labels.

**Comment by tikikun** [comment]
Fixed
